### DISCUSSION TITLE
Did I mess up my keyboard shortcuts in regards to CMD-W (⌘-W) and VSCode?

### DISCUSSION THREAD

**Discussion by danielpietzsch** [blinksh/blink#1650] [Q&A] [answered]
Hi 👋

I recently noticed, that when hitting CMD-W (⌘-W) on my keyboard, Blink will quit completely: like ⌘-Q on a Mac, everything gets closed.

**Previously, I was always able to hit ⌘-W and have a _tab_ closed inside an integrated VSCode window**. Now the same shortcuts quits the app, completely ruining my workflow, as I constantly accidentally quit the app. [**UPDATE**: using the software-keyboard, ⌘-W works as expected: closing just a single VSCode tab at a time.]

I run the latest public iPadOS 16.2 beta (= Beta 4) and Blink v16.0.2.622.

Now, I’m not sure, if this is due to my keyboard shortcuts configuration. But this configuration hasn’t changed in longer while. And I am fairly certain it worked with the iPadOS 16.2 Betas 1 and 2. Not sure about Betas 3 and 4. But maybe that is even unrelated to the Betas!?

However, I’d like to verify that my own shortcuts are not messing with this. But I couldn’t find the default set anywhere. **Can anyone tell me the default set of keyboard shortcuts?** Here’s my current configuration:

![IMG_0190](https://user-images.githubusercontent.com/139080/206255897-01986fdf-c6a8-4035-b917-07232f0df3f8.PNG)

I think I changed “Close tab” from ⌘-W to ⌘-⇧-W in order to have the VSCode ⌘-W-shortcut close a tab. And when I change “Close tab” (back?) to ⌘-W, this indeed closes a _Blink_-Tab again, _not_ quitting the app altogether. Still, I can’t get the ⌘-W-closes-tab-in-VSCode-workflow back again.

Anyone else has the same problem?  
Any hints appreciated!

Maybe this is related? https://switchtoipad.com/features/ipados-16-beta-watch-a-global-keyboard-shortcut-to-close-windows/

**Comment by yury** [comment, marked as answer]
Yes, you are right. New default global cmd+W shortcut cause that.

Since you remove cmd+W shortcut from blink OS captures it. We are thinking about custom shortcuts for vscode.

**Reply by danielpietzsch** [reply]
Thanks very much for the quick reply!

>Since you remove cmd+W shortcut from blink OS captures it. We are thinking about custom shortcuts for vscode.

FWIW, for my personally, it’s be good enough – if possible – to have Blink _always_ capture ⌘-W; no matter the (custom) Shortcuts configuration. Because then I imagine Blink could again forward that shortcut to VSCode. (This of course would then contradict the standard iPadOS behaviour of closing the app altogether.)

**Comment by yury** [comment]
In next release you will be able to map anything to `cmd+w` and `cmd+t` with custom presses.

**Comment by yury** [comment]
**NOTE** for new users

This is a workaround for users who want cmd+w to close VS Code tabs instead of closing a Blink tab. The same instructions apply to `cmd+t`
We are still experimenting with other options, but want you to try this now.

1. In `config` -> `Keyboard` -> `Shortcuts` change `cmd+w` (Close tab) to `cmd+option+w`.
2. In `config` -> `Keyboard` -> `Custom Presses` add new one with `cmd+w` and choose `Press cmd+w`

**Reply by danielpietzsch** [reply]
Thanks so so much! Works perfectly! 🥳

(FYI: one needs the latest version `16.0.3`.)

**Reply by stclj** [reply]
This only works for external keyboards, but sadly not for the on screen keyboard 🙁.

**Reply by grabbou** [reply]
This is great! Wish it was more documented or actually default behavior! Happy to have discovered it!
